ntp-proto/src/packet/mod.rs

```diff
@@ -305,129 +313,136 @@ impl<'a> NtpPacket<'a> {
         let version = (data[0] & 0b0011_1000) >> 3;
 
         match version {
-            3 => {
-                let (header, header_size) =
-                    NtpHeaderV3V4::deserialize(data).map_err(|e| e.generalize())?;
-                let mac = if header_size != data.len() {
-                    Some(Mac::deserialize(&data[header_size..]).map_err(|e| e.generalize())?)
-                } else {
-                    None
-                };
-                Ok((
-                    NtpPacket {
```

Comment:
Good refactor on this match, but again needs more involved review. Could you please make this a separate PR?